src/test/java/de/tum/in/www1/artemis/service/connectors/jenkins/jobs/JenkinsJobPermissionsUtilsTest.java

```diff
@@ -0,0 +1,109 @@
+                      <inheritanceStrategy class="org.jenkinsci.plugins.matrixauth.inheritance.InheritParentStrategy"/>
+                        <permission>hudson.model.Item.Build:instructor1</permission>
+                        <permission>hudson.model.Item.Cancel:instructor1</permission>
+                        <permission>hudson.model.Item.Configure:instructor1</permission>
+                        <permission>hudson.model.Item.Create:instructor1</permission>
+                        <permission>hudson.model.Item.Delete:instructor1</permission>
+                        <permission>hudson.model.Item.Read:instructor1</permission>
+                        <permission>hudson.model.Item.Workspace:instructor1</permission>
+                        <permission>hudson.model.Run.Delete:instructor1</permission>
+                        <permission>hudson.model.Run.Replay:instructor1</permission>
+                        <permission>hudson.model.Run.Update:instructor1</permission>
+                        <permission>hudson.scm.SCM.Tag:instructor1</permission>
+                        <permission>USER:hudson.model.Item.Build:instructor1</permission>
+                        <permission>USER:hudson.model.Item.Cancel:instructor1</permission>
+                        <permission>USER:hudson.model.Item.Configure:instructor1</permission>
+                        <permission>USER:hudson.model.Item.Create:instructor1</permission>
+                        <permission>USER:hudson.model.Item.Delete:instructor1</permission>
+                        <permission>USER:hudson.model.Item.Read:instructor1</permission>
+                        <permission>USER:hudson.model.Item.Workspace:instructor1</permission>
+                        <permission>USER:hudson.model.Run.Delete:instructor1</permission>
+                        <permission>USER:hudson.model.Run.Replay:instructor1</permission>
+                        <permission>USER:hudson.model.Run.Update:instructor1</permission>
+                        <permission>USER:hudson.scm.SCM.Tag:instructor1</permission>
+                    </com.cloudbees.hudson.plugins.folder.properties.AuthorizationMatrixProperty>
+                  </properties>
+                </com.cloudbees.hudson.plugins.folder.Folder>
+                """);
+        final Set<JenkinsJobPermission> allPermissions = Arrays.stream(JenkinsJobPermission.values()).collect(Collectors.toUnmodifiableSet());
+
+        JenkinsJobPermissionsUtils.removePermissionsFromFolder(folderConfig, allPermissions, Set.of("instructor1"));
+
+        final var updatedPermissions = folderConfig.getElementsByTagName("permission");
+        assertThat(updatedPermissions.getLength()).withFailMessage(() -> {
+            try {
+                return "Expected document to contain no permissions:\n" + XmlFileUtils.writeToString(folderConfig);
+            }
+            catch (TransformerException e) {
+                throw new RuntimeException(e);
```

Comment:
Can't we replace this with a fail that prints the stack trace/error message of the caught error message?

Reply:
Simplified in 847b3db1228fd5c7d48562df72abf38e6ffc3e77